ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java

```diff
@@ -259,7 +275,46 @@ public GraphQLDataFetchers(final Set<Capability> supportedCapabilities) {
           ((GraphQLDataFetcherContext) dataFetchingEnvironment.getContext()).getBlockchainQueries();
       final Bytes32 hash = dataFetchingEnvironment.getArgument("hash");
       final Optional<TransactionWithMetadata> tran = blockchain.transactionByHash(Hash.wrap(hash));
-      return tran.map(TransactionAdapter::new);
+      return tran.map(this::getTransactionAdapter);
     };
   }
+
+  private TransactionAdapter getTransactionAdapter(
+      final TransactionWithMetadata transactionWithMetadata) {
+    final Transaction transaction = transactionWithMetadata.getTransaction();
+    return goQuorumPrivacyParameters.isPresent() && transaction.isGoQuorumPrivateTransaction()
+        ? updatePrivatePayload(transaction)
+        : new TransactionAdapter(transactionWithMetadata);
+  }
+
+  private TransactionAdapter updatePrivatePayload(final Transaction currTransaction) {
+    final GoQuorumEnclave enclave = goQuorumPrivacyParameters.get().enclave();
+    final Bytes enclavePayload;
+
+    try {
+      // Retrieve the payload from the enclave
+      enclavePayload =
+          Bytes.wrap(enclave.receive(currTransaction.getPayload().toBase64String()).getPayload());
+
+      // Create a new transaction containing the retrieved payload
+      final Transaction newTransaction =
```

Comment:
I don't like the names currTransaction and newTransaction. Can we make currTransaction -> transaction and make the new Transaction call inline so it doesn't need a name

Reply:
Happy to do that if you find it clearer.